components/accept-financial-contributions/HostsContainer.js

```diff
@@ -0,0 +1,137 @@
+import { Flex } from '@rebass/grid';
+import styled from 'styled-components';
+import { graphql } from '@apollo/react-hoc';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import HostCollectiveCard from './HostCollectiveCard';
+import Loading from '../Loading';
+import HorizontalScroller from '../HorizontalScroller';
+
+import withViewport, { isDesktopOrAbove } from '../../lib/withViewport';
+
+const limit = 12; // nice round number to make even rows of 2, 3, 4
+
+const AllCardsContainer = styled(Flex).attrs({
+  flexWrap: 'wrap',
+  width: '70%',
+  justifyContent: 'space-evenly',
+})``;
+
+const AllCardsContainerMobile = styled(Flex)`
+  overflow-x: auto;
+  scroll-behavior: smooth;
+  width: 100%;
+  padding: 16px;
+`;
+
+const CollectiveCardContainer = styled.div`
+  width: 280px;
+  padding: 20px 15px;
+`;
+
+class HostsContainer extends React.Component {
+  static propTypes = {
+    collective: PropTypes.object,
+    onChange: PropTypes.func,
+    data: PropTypes.object.isRequired,
+    viewport: PropTypes.string,
+    tags: PropTypes.string,
+  };
+
+  render() {
+    const { onChange, data, viewport } = this.props;
+
+    if (!data.hosts || !data.hosts.nodes) {
+      return <Loading />;
+    }
+
+    const hosts = [...data.hosts.nodes];
+
+    return (
+      <Fragment>
+        {!isDesktopOrAbove(viewport) ? (
+          <HorizontalScroller>
+            {ref => (
+              <AllCardsContainerMobile ref={ref}>
+                {hosts.map(host => (
+                  <HostCollectiveCard
+                    key={host.legacyId}
+                    host={host}
+                    collective={this.props.collective}
+                    onChange={onChange}
+                    style={{
+                      flexBasis: 250,
+                      height: 360,
+                      marginRight: 20,
+                      flexShrink: 0,
+                    }}
+                  />
+                ))}
+              </AllCardsContainerMobile>
+            )}
+          </HorizontalScroller>
+        ) : (
+          <AllCardsContainer>
+            {hosts.map(host => (
+              <CollectiveCardContainer key={`${host.legacyId}-container`}>
+                <HostCollectiveCard
+                  key={host.legacyId}
+                  host={host}
+                  collective={this.props.collective}
+                  onChange={onChange}
+                />
+              </CollectiveCardContainer>
+            ))}
+          </AllCardsContainer>
+        )}
+      </Fragment>
+    );
+  }
+}
+
+const getHostsQuery = gqlV2`
+query getHosts($tags: [String], $limit: Int) {
+  hosts(tags: $tags, limit: $limit) {
+    totalCount
+    nodes {
+      id
+      legacyId
+      createdAt
+      settings
+      type
+      name
+      slug
+      description
+      currency
+      totalHostedCollectives
+      hostFeePercent
+      stats {
+        yearlyBudget {
+          value
+        }
+      }
+      members {
```

Comment:
This was note working well when I checked locally (API error).

Also, it's not the right value for what we want to display. I'm working on the right GraphQL property.